zabapgit.prog.abap

```diff
@@ -11476,6 +11476,344 @@ CLASS lcl_object_sfbs IMPLEMENTATION.
 
 ENDCLASS.                    "lcl_object_SFBS IMPLEMENTATION
 
+*----------------------------------------------------------------------*
+*       CLASS lcl_object_W3SUPER DEFINITION
+*----------------------------------------------------------------------*
+*   Web Reporting/Internet Transaction Server MIME Types (super class)
+*----------------------------------------------------------------------*
+CLASS lcl_object_w3super DEFINITION INHERITING FROM lcl_objects_super ABSTRACT.
+
+  PUBLIC SECTION.
+    INTERFACES lif_object.
+
+ENDCLASS. "lcl_object_W3SUPER DEFINITION
+
+*----------------------------------------------------------------------*
+*       CLASS lcl_object_W3SUPER IMPLEMENTATION
+*----------------------------------------------------------------------*
+*   Web Reporting/Internet Transaction Server MIME Types (super class)
+*----------------------------------------------------------------------*
+CLASS lcl_object_w3super IMPLEMENTATION.
+
+  METHOD lif_object~jump.
+    " No idea how to just to SMW0
+    _raise 'Please go to SMW0 for W3MI object'.
+  ENDMETHOD.                    "jump
+
+  METHOD lif_object~get_metadata.
+    rs_metadata = get_metadata( ).
+  ENDMETHOD.
+
+  """"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+  " W3xx EXISTS
+  """"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+  METHOD lif_object~exists.
+    DATA lv_objid TYPE wwwdata-objid.
+
+    SELECT SINGLE objid into lv_objid
+      FROM wwwdata
+      WHERE relid = ms_item-obj_type+2(2)
+      AND   objid = ms_item-obj_name
+      AND   srtf2 = 0.
+
+    IF lv_objid IS INITIAL.
+      RETURN.
+    ENDIF.
+
+    rv_bool = abap_true.
+
+  ENDMETHOD.
+
+  """"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+  " W3xx SERIALIZE
+  """"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+  METHOD lif_object~serialize.
+    DATA ls_key      TYPE wwwdatatab.
+    DATA lt_w3mime   TYPE STANDARD TABLE OF w3mime.
+    DATA lt_w3html   TYPE STANDARD TABLE OF w3html.
+    DATA lt_w3params TYPE STANDARD TABLE OF wwwparams.
+    DATA l_xstring   TYPE xstring.
+    DATA ls_wwwparam TYPE wwwparams.
+    DATA l_size      TYPE int4.
+    DATA l_base64str TYPE string.
+    DATA lo_utility  TYPE REF TO cl_http_utility.
+
+    ls_key-relid = ms_item-obj_type+2(2).
+    ls_key-objid = ms_item-obj_name.
+
+    SELECT SINGLE * INTO CORRESPONDING FIELDS OF ls_key
+      FROM wwwdata
+      WHERE relid = ls_key-relid
+      AND   objid = ls_key-objid
+      AND   srtf2 = 0.
+
+    IF sy-subrc IS NOT INITIAL.
+      RETURN.
+    ENDIF.
```

Comment:
looks like the above statements can be replaced with calling method lif_object~exists instead?